linode/manager
# upcoming: [M3-7753] - Tag custom analytics events for account switching

## Description 📝
For Parent/Child account switching flows, we want to keep track of:

- the number of users who click the Switch Account buttons in the user menu and Account Landing page
- the number of proxy users who switch back to parent accounts via the Switch Account drawer
- the number of proxy users who encounter the session expiry modal when a parent's token has expired

## Changes  🔄
List any change relevant to the reviewer.
- Adds the `sendSwitchAccountEvent`, `sendSwitchToParentAccountEvent`, and `sendSwitchAccountSessionExpiryEvent` functions to `analytics.ts` and calls them when the corresponding button clicks are made.

## Preview 📷
| Component  | Custom Event   |
| ------- | ------- |
| ![SwitchAccountButton-UserMenu](https://github.com/linode/manager/assets/114685994/195542a1-dd01-40f8-aa36-f86ba5ce60aa) | ![Screenshot 2024-02-13 at 8 59 59 AM](https://github.com/linode/manager/assets/114685994/5f35de70-84c0-4f75-a4b8-77b522ca681e) |
| ![SwitchAccountButton-AccountLanding](https://github.com/linode/manager/assets/114685994/79a494e5-1dc6-47c1-9771-20d03b02dd3b) | ![Screenshot 2024-02-13 at 9 00 19 AM](https://github.com/linode/manager/assets/114685994/ac44e615-d9e7-412e-b63d-1b71274bbbe5) |
| ![SwitchAccountDrawer-ProxyToParent](https://github.com/linode/manager/assets/114685994/b0721ae6-fbf6-49f8-8d44-b90709412a96) | ![Screenshot 2024-02-13 at 9 05 31 AM](https://github.com/linode/manager/assets/114685994/6e393b9a-4b34-430e-965c-d3b98ab0159a) |
| ![Screenshot 2024-02-13 at 9 17 32 AM](https://github.com/linode/manager/assets/114685994/454defd2-6073-44d6-a516-8906356c62a9) | ![Screenshot 2024-02-13 at 9 01 02 AM](https://github.com/linode/manager/assets/114685994/ed40371a-dcab-49b7-b181-ef579060d07d) ![Screenshot 2024-02-13 at 9 02 24 AM](https://github.com/linode/manager/assets/114685994/b635c9b0-f184-47ad-8d72-de53e9388dd8) |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- Check out this PR and `yarn dev`
- Confirm the parent/child feature flag is on and mocks are on
- Go to http://localhost:3000 and turn on analytics logs for Adobe: 
   - In the browser console, type `_satellite.setDebug(true)`
- Confirm that the `user_type` for `*/profile` is `"parent"` in [`serverHandlers.ts`](https://github.com/linode/manager/blob/aa9899ce8a88950ff83a7198c4d0033f0bb92691/packages/manager/src/mocks/serverHandlers.ts#L556)

### Verification steps 
(How to verify changes)
- Go to http://localhost:3000/account/billing
- Open the browser console and click on the Switch Account button 
- Confirm the `sendSwitchAccountEvent` has fired in the browser console log with label `Account Landing`
- Open the user menu and click the Switch Account button 
- Confirm the `sendSwitchAccountEvent` has fired in the browser console log with label `User Menu`
- Change the `user_type` for `*/profile` to `"proxy"` in [`serverHandlers.ts`](https://github.com/linode/manager/blob/aa9899ce8a88950ff83a7198c4d0033f0bb92691/packages/manager/src/mocks/serverHandlers.ts#L556)
- Click the Switch Account button and click "switch back to parent account". 
- Confirm the `sendSwitchToParentAccountEvent` has fired in the browser console log
- Clear the application data in the browser console so that no parent token is cached in local storage
- Click the Switch Account button and observe the session expiry dialog pops up.
- Confirm that clicking any of the buttons to dismiss the dialog fires a `sendSwitchAccountSessionExpiryEvent` with the appropriate label. Note that `Log In` is tricky because the event will only be visible for a brief moment before the page redirects.

## As an Author I have considered 🤔

*Check all that apply*
